server/vcr-server/agent_webhooks/views.py

```diff
@@ -464,3 +410,333 @@ def handle_register_issuer(message):
+                    "attach_id": "0",
+                    "format": "hlindy-zkp-v1.0"
+                }
+            ],
+            "credential_preview": {
+                "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/issue-credential/2.0/credential-preview",
+                "attributes": [
+                    {
+                        "name": "reservation_number",
+                        "mime-type": "text/plain",
+                        "value": "000003"
+                    },
+                    {
+                        "name": "guest_name",
+                        "mime-type": "text/plain",
+                        "value": "Giuseppe Verde"
+                    },
+                    {
+                        "name": "reservation_date",
+                        "mime-type": "text/plain",
+                        "value": "2021-04-01"
+                    },
+                    {
+                        "name": "reservation_date_cancelled",
+                        "mime-type": "text/plain",
+                        "value": "2021-04-02"
+                    },
+                    {
+                        "name": "another_date",
+                        "mime-type": "text/plain",
+                        "value": "2021-04-05"
+                    },
+                    {
+                        "name": "num_guests",
+                        "mime-type": "text/plain",
+                        "value": "4"
+                    }
+                ]
+            }
+        },
+        "auto_issue": False
+    }
+    """
+
+    cred_ex_id = message["cred_ex_id"]
+    LOGGER.debug(f'Credential: state="{state}" cred_ex_id="{cred_ex_id}"')
+    response_data = {}
+
+    try:
+        if state == "offer-received":
+            LOGGER.debug("After receiving credential offer, send credential request")
+            # no need to perform a task, we run the agent with the --auto-respond-credential-offer flag set
+            response_data = {
+                "success": True,
+                "details": f"Received offer on credential exchange {cred_ex_id}",
+            }
+
+        elif state == "credential-received":
+            cred_issue = message["cred_issue"]
+            if cred_issue is None:
+                LOGGER.error(" >>> Error cred_issue missing for " + cred_ex_id)
+                return Response("Error cred_issue missing for credential", status=status.HTTP_400_BAD_REQUEST)
+
+            # You can include this exception to test error reporting
+            if RANDOM_ERRORS:
+                if 1 == random.randint(1, 50):
+                    print("Raise exception 1 for " + cred_ex_id)
+                    raise Exception("Depliberate error to test problem reporting 1")
+                if 1 == random.randint(1, 50):
+                    print("Return processing error 1 for " + cred_ex_id)
+                    return Response("Deliberate error to test bad request 1", status=status.HTTP_400_BAD_REQUEST)
```

Comment:
It would be nice to pull this random error function into a utility, rather than having duplicated code across different handlers